ert_shared/plugins/plugin_manager.py

```diff
@@ -179,14 +215,23 @@ def _setup_temp_environment_if_not_already_set(self, env):
             if self.backup_env.get(name) is None:
                 if value is not None:
                     os.environ[name] = value
+                    logging.debug("Set environment variable {}={}".format(name, value))
+            else:
+                logging.debug(
+                    "Environment variable already set {}={}".format(
+                        name, self.backup_env.get(name)
+                    )
+                )
 
     @python3only
     def _reset_environment(self):
         for name, value in self.env.items():
             if self.backup_env.get(name) is None and name in os.environ:
                 del os.environ[name]
+                logging.debug("Reset environment variable {}".format(name))
 
     @python3only
     def __exit__(self, *args):
         self._reset_environment()
         shutil.rmtree(self.tmp_dir)
+        logging.debug("Removed tmp dir for site-config")
```

Comment:
Not very important, but a couple of thoughts regarding logging. I've sometimes found it useful having a log statement in front of the fan logging which shit is going to hit soon. That way it is easier to pinpoint what went wrong. In this case i would have put the log statement right in front of rmtree and logged "Removed tmpdir {} for site-config".format(self.tmp_dir) . If it is necessary... I don't know. By the stacktrace you would still have which line failed, so i guess it's fine as it is - you can decide!

Reply:
Agreed! I have added comments before some of the actions, but also kept some of the post action logging so that we can see that the side-effect actually occurred.